flow/project.py

```diff
@@ -1048,12 +1049,9 @@ def _generate_id(self, jobs, operation_name=None, index=0):
             concat_jobs_str = str(jobs[0])[0:8]
 
         separator = getattr(project._environment, 'JOB_ID_SEPARATOR', '/')
-        readable_name = '{project}{sep}{jobs}{sep}{op_string}{sep}{index:04d}{sep}'.format(
-                    sep=separator,
-                    project=str(project)[:12],
-                    jobs=concat_jobs_str,
-                    op_string=op_string[:12],
-                    index=index)[:max_len]
+        readable_name = f'{str(project)[:12]}{separator}{self.name}{separator}' \
+                        f'{len(jobs)}{separator}{concat_jobs_str}{separator}{index:04d}' \
+                        f'{separator}'[:max_len]
```

Comment:
I've stared at this for a while and I believe this should be fine. It would have been helpful to add a sample of this to the review comments.

Reply:
@csadorf yes, I should have posted a sample so that it becomes easy for a review. I apologise.

Reply:
Also, if you're talking about the jobs passed in the operation function then yes I agree that users could do that but if we provide a little information from our side then I guess it'll save an extra print statement from the users. Please suggest.

Reply:
Sorry, I should have phrased that differently: It's always helpful to provide some samples etc., but absolutely no need to apologize. 👍